invenio_records_resources/services/records/service.py

```diff
@@ -552,3 +558,37 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def reindex_latest_first(
+        self, identity, search_preference=None, extra_filter=None, uow=None
+    ):
+        """Reindexes records matching the query filter, prioritizing latest versions.
+
+        Records will be retrieved from search engine and current versions re-indexed first.
+        The latest version of a record is reindexed twice, but we trade that off for the guarantee that it also gets re-indexed first.
+        """
+        self.require_permission(identity, "manage")
```

Comment:
Minor: the permission of `reindex` is `search`, and not `manage`. We might want to align the 2? `manage` actually looks better, because we should re-index when something changed. The only exception is the `relation update`: when a related record is updated, it can trigger a re-index (probably with system process).

Reply:
I had this question myself. However, since we are allowing the reindexing of records (any records) I thought it would be better to have it more restricted.  In reality, the `search` object will only allow the caller to reindex records that they can `read`, so it wouldn't be too bad to allow e.g. a user to reindex their own records.